core/corehttp/gateway_handler.go

```diff
@@ -81,7 +81,7 @@ func (i *gatewayHandler) resolveNamePath(ctx context.Context, p string) (string,
 	if strings.HasPrefix(p, IpnsPathPrefix) {
 		elements := strings.Split(p[len(IpnsPathPrefix):], "/")
 		hash := elements[0]
-		rp, err := i.node.Namesys.Resolve(ctx, hash)
+		rp, err := i.node.Namesys.Resolve(ctx, hash, 0)
```

Comment:
don't think we should resolve names infinitely. we should have a max, something like `16` or `32`. otherwise we can hose machines by setting up very long chains and causing nodes to resolve for a very long time). (maybe infinitely, if we can generate + publish names faster than the nodes time out)